inter-operator/architecture/basic.md

```diff
@@ -531,6 +545,73 @@ status:
+#### Service Instance Reconciler
+Service Instance Reconciler is the custom controller which watches across multiple clusters, part of the cluster registry(set of `SFClusters`) and reconciles a `SFServiceInstance` between master cluster and its assigned cluster, assigned by [Scheduler](#schedulers).
+#### Service Binding Reconciler
+Service Binding Reconciler is the custom controller which watches across multiple clusters, part of the cluster registry(set of `SFClusters`) and reconciles a `SFServiceBinding` between master cluster and its assigned cluster, assigned by [Scheduler](#schedulers).
+### Schedulers
+Schedulers are basically custom controller running on master cluster watching on `SFServiceInstances` and schedules/assigns them `clusterId` where the instance need to be provisioned, depending on the scheduling algorithm it implements. We currently have implemented the following set of schedulers described below. Activating a scheduler is config driven to be passed when someone deploys Inter-operator.
+#### DefaultScheduler
+This is just a sample scheduler if one does not want multiple clusters in the setup. In that case, it schedules all the instances in the one cluster which is part of the setup.
+#### Round Robin Scheduler
+As the name suggests, round robin scheduler schedules instances in round robin fashion. At this point, it does not take care of capacity and if interoperator restarts, it starts scheduling from the beginning. 
+#### Least filled Scheduler
+This scheduler schedules instance in the least filled cluster.
+### Provisioner
+Provisioner was also already introduced earlier, please read about it in the earlier section [here](#service-fabrik-inter-operator-provisioner)
+## Deployment Flow
+Following are the flow for a deployment of Interoperator.
+1. When Interoperator is deployed initially, one deploys the [broker](#broker), [MultiClusterDeployer](#multiclusterdeployer) and the [Scheduler](#schedulers) component in a cluster, called as master cluster.
+2. After this, the operator should create the `SFServices`, `SFPlans` and `SFClusters` in the master cluster. `SFClusters` is simply the list/registry of all clusters where you want to provision the instances. We also refer to them as sister cluster interchangebly. Master cluster can also be part of the cluster registry and be a sister cluster in itself, if someone wants to use it for service provisioning as well.
+3. [Provisioner Controller](#provisioner-controller) then takes care of replicating the provisioner component to all sister clusters and [Service Replicator](#service-replicator) takes care of replicating the SFServices and SFPlans in all the clusters.
+
+Now the setup is ready for taking requests. We depict this in the picture below.
+![Inter-operator Deployment Flow](images/Deployment%20Flow.png)
```

Comment:
Can we visualize controllers and CRDs differently?

Reply:
With different color coding ? Or shall I put them inside APIServer shown as a state ?

Reply:
Made some changes in the diagrams. WDYTN?

Reply:
Color-coding as well as different shape? Or as you mentioned, inside the APIServer as a state.

Reply:
I changed the color coding and also kept them outside the interoperator box to denote that they are not a sub-component. Let me know if it looks good now.